plugins/flytekit-spark/flytekitplugins/spark/models.py

```diff
@@ -28,6 +29,10 @@ def __init__(
         databricks_conf: Dict[str, Dict[str, Dict]] = {},
         databricks_token: Optional[str] = None,
         databricks_instance: Optional[str] = None,
+        driver_pod_template: Optional["PodTemplate"] = None,
```

Comment:
as much as we want to get rid of model files, since these are there, i say we stick to them for now.  There's a [K8sPod model class](https://github.com/flyteorg/flytekit/blob/0e82799e3a701075012a6da76f4a4f5846d5c969/flytekit/models/task.py#L870) that I think we should use here since the [idl is going](https://github.com/flyteorg/flyteidl/pull/450/files#diff-394e059361a84ee9e734be9a23262106290623951064a5bb44ab92b6fb86cbe6R39) to use that object too.